packages/airnode-deployer/src/handlers/aws/index.ts

```diff
@@ -104,6 +104,17 @@ export async function processHttpRequest(
     format: parsedConfig.nodeSettings.logFormat,
     level: parsedConfig.nodeSettings.logLevel,
   });
+
+  // Check if the request origin header is allowed in the config
+  const originVerification = verifyRequestOrigin(
+    parsedConfig.nodeSettings.httpGateway.enabled ? parsedConfig.nodeSettings.httpGateway.corsOrigins : [],
+    event.headers['origin']
+  );
+  // Respond to preflight requests if the origin is allowed
+  if (originVerification.success && event.httpMethod === 'OPTIONS') {
+    return { statusCode: 204, headers: originVerification.headers, body: '' };
```

Comment:
I like 204 status code for success response.

However, we should also return early if there is an error. I suggest status code `400` with some message.

Reply:
By error do you mean that the origin is not allowed? The browser handles the CORS not allowed error if the correct headers are not returned, but do you mean that we should have a custom error for this case?

Reply:
If the request is OPTIONS we should not even attempt to call lambda, just return the headers to the browser.

Reply:
I meant the Airnode handlers, but I chose poor words to decscribe this. I mean basically the same thing as in the other comment, so we can resolve this one.